snp_motion_execution/launch/ros2_control.launch.py

```diff
@@ -39,12 +42,18 @@ def launch(context, *args, **kwargs):
     with open(controllers_file.perform(context), 'r') as f:
         controllers = yaml.safe_load(f)
 
+    # define controller manager executable name based on ROS_DISTRO
+    if ros_distro > "foxy":
+        exec_name = "spawner"
+    else:
+        exec_name = "spawner.py"
+
     for key, val in controllers['controller_manager']['ros__parameters'].items():
         if type(val) is dict and type(val['type']) is str:
             nodes.append(
                 Node(
                     package="controller_manager",
-                    executable="spawner.py",
+                    executable=exec_name,
```

Comment:
I think you can do this in a one-liner instead

```suggestion
                    executable='spawner' if os.environ['ROS_DISTRO'] > 'foxy' else 'spawner.py',
```